app/react/Thesauris/components/ThesauriFormGroup.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Field } from 'react-redux-form';
+import { DragAndDropContainer } from 'app/Layout/DragAndDrop';
+import { Icon } from 'UI';
+
+import FormGroup from 'app/DocumentForm/components/FormGroup';
+
+import ThesauriFormField from './ThesauriFormField';
+
+export class ThesauriFormGroup extends Component {
+  constructor(props) {
+    super(props);
+    this.focus = () => { this.groupInput.focus(); };
+    this.renderItem = this.renderItem.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.removeGroup = this.removeGroup.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps) {
```

Comment:
@habbes we should not use shouldComponentUpdate, if this is for performance reasons consider making the component pure, connect()(Component) we should have a pure() function to do this actually, so make component pure and remove shouldComponentUpdate, also running a JSON.stringify on a shouldComponentUpdate function its going to be not that good performance wise as it can be really really slow and this is a function that gets called always before render.

Reply:
Yeah, I think they were there for performance reasons.